jabgui/src/main/java/org/jabref/gui/newentry/NewEntryViewModel.java

```diff
@@ -130,6 +144,34 @@ public NewEntryViewModel(GuiPreferences preferences,
         bibtexWorker = null;
     }
 
+    public void populateDOICache() {
+        doiCache.clear();
+        Optional<BibDatabaseContext> activeDatabase = stateManager.getActiveDatabase();
+
+        activeDatabase.ifPresent(bibDatabaseContext -> bibDatabaseContext.getEntries()
```

Comment:
optionals also support getting a streamm
```suggestion
  activeDatabase.stream()
                      .map(BibDatabaseContext::getEntries)
                      .flatMap(Collection::stream)
                      .forEach(bibEntry -> bibEntry.getField(StandardField.DOI)
                                                   .ifPresent(doi -> doiCache.put(doi, bibEntry)));
```

Reply:
Thank you for the suggestion. I did not know that !

Reply:
In hindsight that kind of makes sense, actually. It is an optional after all.